fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanAssembler.java

```diff
@@ -835,6 +835,12 @@ public Map<String, Object> updateFrom(JsonCommand command, Loan loan) {
             loan.updateLoanRates(rateAssembler.fromParsedJson(command.parsedJson()));
         }
 
+        if (command.hasParameter(LoanProductConstants.ENABLE_DOWN_PAYMENT)) {
```

Comment:
Please follow the same logic as it was done when loan was submitted:
- You can disable down payment on loan, when the it was allowed on the loan product, but not the otherwise!

Reply:
Done! new validation method implemented in the LoanScheduleValidator